internal/controller/controllers/agent_controller.go

```diff
@@ -610,6 +609,40 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 	return ret, nil
 }
 
+func (r *AgentReconciler) UpdateDay2InstallPogress(ctx context.Context, h *models.Host, agent *aiv1beta1.Agent, spokeClient spoke_k8s_client.SpokeK8sClient) (ctrl.Result, error) {
```

Comment:
> the update of the stage can be in the same place for day1 and day2

The day1 host update is determined by the installer running on the node and later by the assisted-installer-controller running inside the installed cluster.
Day2 host installation progress is different because we don't have the assisted-installer-controller to report the host progress, hence we need this code...

Reply:
This is correct, but this is already set by the code that changes the stage in the database.  Therefore, the stage is Kube API is just a reflection of what exists in the database.

Reply:
I agree that it's better, but I'm concerned that this will lead to a UT changes nightmare.
I can give it a try in another PR

Reply:
Hmm, actually in order for the stage in Kube API to reflect the stage we have in the database.
I'll need to fetch the agent from the DB again (or set it manually on the host object here).
I think there isn't much value in it, and perhaps the current implementation is clearer - in the case of day2 host we set the installation stage here (with some logic behind it) in the case of day 1 we just copy it from the host (under the `else` condition).